character.cpp

```diff
@@ -17,7 +17,13 @@ void Character::reduceHealthByDamage(const Character &attacker) {
 
 
 Character Character::parseUnit(const std::string& fname) {
-	std::map<std::string, std::string> P = Parser::jsonParser(fname);
+	std::map<std::string, std::string> P;
+	if (fname.find(".json", fname.size() - 5) == -1){
+		P = Parser::StringToMap(fname);
+	}
+	else{
+		P = Parser::jsonParser(fname);
+	} 

```

Comment:
Miért ragaszkodsz ennyire hozzá, hogy valahol ez az if szerepeljen? Kéri valahol a feladat, hogy legyen egy függvény, ami képes detektálni a paraméter tartalma alapján, hogy az egy fájl neve, vagy egy JSON tartalom, és ennek megfelelően kell működnie?
A karakternek egyelőre elég, ha egy fájlnév (ld. parancssori paraméter) alapján tudja kérni a parzert az adatok betöltésére. A parzer meg most kicsit túl általános is, de ez nem baj, később esetleg át lehet majd tenni egy másik programba, ahol meg a többi függvényére lesz szükség.